### PR TITLE
Web viewer profiling and optimizations

### PR DESCRIPTION
**Pull request by kwokcb**
## Updates
- Fix to avoid running 2 instance of the viewer as `HtmlWebpackPlugin` will inject code already to run an instance. 
Loading the main code again results in 2 competing instances being executed affecting all load / update time. (`index.ejs`)
- Avoid translating 4 channel IBLs again on load. Remapping is only required if is 3 channel of which the packaged IBL is not. (`helpers.js`)
- Avoid unnecessary geometry transform updates on every frame. (`viweer.js`)
- Restore texture uniform asynchronous load. This still appears to be the most costly in terms of material setup performance.
- Add some perf logging.

### CODE REVIEW

File: javascript/MaterialXView/index.ejs [hunk 1]
```diff
@@ -37,6 +37,5 @@
       <canvas id="webglcanvas" tabindex="1" style="outline: none;"></canvas>
     </div>
     <script src="JsMaterialXGenShader.js"></script>
-    <script src="main.js"></script>
```
Comment (kwokcb) [inline javascript/MaterialXView/index.ejs:40]: This should never have been there as it invokes 2 instances to run causing all evaluation and load to occur twice.

Comment (jstone-lucasfilm) [reply javascript/MaterialXView/index.ejs:40]: Good catch, @kwokcb!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -215,6 +215,7 @@ jobs:
     - name: Shader Validation Tests
       if: matrix.test_shaders == 'ON' && runner.os == 'Windows'
       run: |
+        vcpkg/vcpkg remove glslang
```
Comment (jstone-lucasfilm) [inline .github/workflows/main.yml:218]: Now that the glslang test is disabled, I believe we can remove this workaround from recent pull requests.

File: javascript/MaterialXView/source/helper.js [hunk 3]
```diff
@@ -38,34 +38,41 @@ export function prepareEnvTexture(texture, capabilities)
...
-        case THREE.FloatType:
-            rgbaData = new Float32Array(length);
-            break;
-        case THREE.HalfFloatType:
-            rgbaData = new Uint16Array(length);
-            break;
-        default:
-          break;
-    }
+            case THREE.FloatType:
```
Comment (jstone-lucasfilm) [inline javascript/MaterialXView/source/helper.js:54]: Is this change intentional, or just an accidental whitespace edit?

File: javascript/MaterialXView/source/helper.js [hunk 4]
```diff
@@ -38,34 +38,41 @@ export function prepareEnvTexture(texture, capabilities)
...
 
     if (rgbaData)
     {
         for (let i = 0; i < length / 4; i++)
         {
-            rgbaData[(i * 4) + 0] = rgbData[(i * 3) + 0];
-            rgbaData[(i * 4) + 1] = rgbData[(i * 3) + 1];
-            rgbaData[(i * 4) + 2] = rgbData[(i * 3) + 2];
-            rgbaData[(i * 4) + 3] = 1.0;
+                rgbaData[(i * 4) + 0] = rgbData[(i * 3) + 0];
```
Comment (jstone-lucasfilm) [inline javascript/MaterialXView/source/helper.js:68]: As above, is this an intentional change to whitespace?

**Review by jstone-lucasfilm — COMMENTED** [review]
These optimizations and timing tests look really promising, thanks @kwokcb, and I have just a few thoughts posted below.

**Review by jstone-lucasfilm — APPROVED** [review]
Thanks for these optimizations and improvements, @kwokcb!
